crates/worker/src/partition/mod.rs

```diff
@@ -523,6 +551,202 @@ where
+            ) => {
+                self.leadership_state
+                    .self_propose_and_wait_response(
+                        service_invocation.partition_key(),
+                        Command::Invoke(service_invocation),
+                        response_tx,
+                    )
+                    .await;
+            }
+            PartitionProcessorRpcRequestInner::AppendInvocation(
+                mut service_invocation,
+                AppendInvocationReplyOn::Submitted,
+            ) => {
+                service_invocation.submit_notification_sink =
+                    Some(SubmitNotificationSink::Ingress { request_id });
+
+                self.leadership_state
+                    .handle_rpc_proposal_command(
+                        request_id,
+                        response_tx,
+                        service_invocation.partition_key(),
+                        Command::Invoke(service_invocation),
+                    )
+                    .await
+            }
+            PartitionProcessorRpcRequestInner::AppendInvocation(
+                mut service_invocation,
+                AppendInvocationReplyOn::Output,
+            ) => {
+                service_invocation.response_sink =
+                    Some(ServiceInvocationResponseSink::Ingress { request_id });
+
+                self.leadership_state
+                    .handle_rpc_proposal_command(
+                        request_id,
+                        response_tx,
+                        service_invocation.partition_key(),
+                        Command::Invoke(service_invocation),
+                    )
+                    .await
+            }
+            PartitionProcessorRpcRequestInner::GetInvocationOutput(
+                invocation_query,
+                GetInvocationOutputResponseMode::BlockWhenNotReady,
+            ) => {
+                // Try to get invocation output now, if it's ready reply immediately with it
+                if let Ok(ready_result @ PartitionProcessorRpcResponse::Output(_)) = self
+                    .handle_rpc_get_invocation_output(
+                        request_id,
+                        invocation_query.clone(),
+                        partition_store,
+                    )
+                    .await
+                {
+                    self.respond_to_rpc(response_tx.prepare(Ok(ready_result)));
+                    return;
+                }
+
+                self.leadership_state
+                    .handle_rpc_proposal_command(
+                        request_id,
+                        response_tx,
+                        invocation_query.partition_key(),
+                        Command::AttachInvocation(AttachInvocationRequest {
+                            invocation_query,
+                            response_sink: ServiceInvocationResponseSink::Ingress { request_id },
+                        }),
+                    )
+                    .await
```

Comment:
I think here is a potential for a race condition. Assume that the invocation is not ready but we enqueue the `AttachInvocation` after an `End` command. In this case, we won't respond with the result if it is not an idempotent invocation.

Reply:
I think this is not a real problem so please ignore this comment.